include/revng/Support/GraphAlgorithms.h

```diff
@@ -443,42 +514,62 @@ getBackedges(GraphT Block,
     for (NodeRef Succ :
          llvm::make_range(GT::child_begin(Block), GT::child_end(Block))) {
       if (State.onStack(Succ)) {
-        State.insertBackedge(Block, Succ);
+        Backedges.insert(EdgeDescriptor(Block, Succ));
       }
     }
   }
 
-  return State.getBackedges();
+  return Backedges;
 }
 
 template<class GraphT, class GT = llvm::GraphTraits<GraphT>>
-llvm::SmallSet<revng::detail::EdgeDescriptor<typename GT::NodeRef>, 4>
+llvm::SmallSetVector<revng::detail::EdgeDescriptor<typename GT::NodeRef>, 4>
 getBackedges(GraphT Block) {
-  return getBackedges(Block, [](const typename GT::NodeRef B) { return true; });
+  llvm::SmallPtrSet<typename GT::NodeRef, 4> EmptySet;
+  return getBackedgesImpl<revng::detail::FilterSet::BlackList>(Block, EmptySet);
 }
 
 template<class GraphT, class GT = llvm::GraphTraits<GraphT>>
-llvm::SmallPtrSet<typename GT::NodeRef, 4>
+llvm::SmallSetVector<revng::detail::EdgeDescriptor<typename GT::NodeRef>, 4>
+getBackedgesWhiteList(GraphT Block,
+                      llvm::SmallPtrSet<typename GT::NodeRef, 4> &Set) {
+  return getBackedgesImpl<revng::detail::FilterSet::WhiteList>(Block, Set);
+}
+
+template<class GraphT, class GT = llvm::GraphTraits<GraphT>>
+llvm::SmallSetVector<revng::detail::EdgeDescriptor<typename GT::NodeRef>, 4>
+getBackedgesBlackList(GraphT Block,
+                      llvm::SmallPtrSet<typename GT::NodeRef, 4> &Set) {
+  return getBackedgesImpl<revng::detail::FilterSet::BlackList>(Block, Set);
+}
+
+// TODO: remove the `IgnoreList` parameter once the new AVI implementation is
+//       merged, checking that no other uses are still present
+template<class GraphT, class GT = llvm::GraphTraits<GraphT>>
+llvm::SmallSetVector<typename GT::NodeRef, 4>
 nodesBetweenNewImpl(GraphT Source,
                     GraphT Target,
-                    const llvm::SmallPtrSetImpl<GraphT> *IgnoreList = nullptr) {
+                    const llvm::SmallSetVector<GraphT, 4> *IgnoreL = nullptr) {
   using NodeRef = typename GT::NodeRef;
-  using StateType = revng::detail::DFSReachableState<NodeRef>;
+  using StateType = revng::detail::DFStackReachable<NodeRef>;
   StateType State;
 
+  using SmallSetVector = llvm::SmallSetVector<NodeRef, 4>;
+
   // If the `IgnoreList` is not empty, populate the ext set with the nodes that
-  // it contains.
-  if (IgnoreList != nullptr) {
-    for (GraphT Element : *IgnoreList) {
+  // it contains
+  if (IgnoreL != nullptr) {
+    for (GraphT Element : *IgnoreL) {
```

Comment:
No abbreviations

Reply:
I guess this was due to an outdated commit, the filter parameter is now called `Set` for the `nodesBetweenImpl` function and friends, and this change has been introduced when dropping the old `nodesBetween` in 510afd77cd288a2f899ef0afaa53590047c584c2.